partiql-lang/src/main/kotlin/org/partiql/lang/planner/GlobalVariableResolver.kt

```diff
@@ -17,6 +18,21 @@ sealed class GlobalResolutionResult {
      */
     data class GlobalVariable(val uniqueId: String) : GlobalResolutionResult()
 
+    /**
+     * A success case. Refers to a variable that is contained within a namespace (AKA catalog/schema).
+     */
+    abstract class NamespacedVariable(
+        val uniqueId: String
+    ) : GlobalResolutionResult() {
+        /**
+         * When attempting to resolve a qualified identifier, say `FROM cat1.schema1.table1.attr1.attr2`, the [GlobalVariableResolver]
+         * _may_ resolve only the first few steps of the identifier, say `cat1.schema1.table1`. Therefore, it must
+         * return the resolved [uniqueId] (say `cat1:::schema1:::table1`) and the remaining steps `attr1.attr2` in order for the planner
+         * to convert these to path expressions.
+         */
+        abstract fun getRemainingSteps(): List<BindingName>
```

Comment:
IMHO this might be a little too forward thinking... i.e. trying to design for a future we know little about.  Or is there more to this than I am aware of?  

In my case, all parts of a table identifier will be completely resolved or we will abort w/undefined table error.  This *might* make sense in the event that we have only part of the schema... but that is not in my foreseeable future. I suggest dropping this.

If this is dropped, I don't think we need a distinction between a `GlobalVariable` and a `NamespacedVariable`  either... `GlobalVariable` will still be fine for our case...  Even though we're kind of using it in a way not entirely consistent with the name.  Maybe, "ResolvedVariable" is a better name?

Reply:
Thinking again about this... was the intention to handle the scenario when some of the parts of the qualified identifier could be resolved but others not?  I guess I'm not 100% sure I understand the intent here.

Reply:
> Thinking again about this... was the intention to handle the scenario when some of the parts of the qualified identifier could be resolved but others not?

Yes, that is the intention. In the `INSERT INTO <table name>` use-case, you wouldn't allow `remainingSteps` to have anything. But, in a query such as the one mentioned in the Javadocs, it is useful to know what is actually a global vs what is a path expression. This allows for the engine-implementer to enforce the accuracy of the path expression and for the global-variable-resolver implementer to only take care of returning "global" variables.

This is something that has been added to `v1`.